x/auth/client/rest/sign.go

```diff
@@ -30,21 +26,34 @@ func SignTxRequestHandlerFn(cdc *codec.Codec, cliCtx context.CLIContext) http.Ha
 	return func(w http.ResponseWriter, r *http.Request) {
 		var m SignBody
 
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
+		if err := utils.ReadRESTReq(w, r, cdc, &m); err != nil {
 			utils.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
 			return
 		}
-		err = cdc.UnmarshalJSON(body, &m)
-		if err != nil {
+
+		if !m.BaseReq.ValidateBasic(w) {
+			return
+		}
+
+		// validate tx
+		// discard error if it's CodeUnauthorized as the tx comes with no signatures
```

Comment:
I'm a little bit confused as to why CodeUnautherized implies that there tx comes with no signatures... maybe we should have `CodeNoSignatures`? Don't deal with this code a lot just an outsider's perspective

Reply:
`CodeNoSignatures` sounds good and self-explanatory to me :+1:

Reply:
Amended, please review again